Cython/Compiler/Optimize.py

```diff
@@ -703,6 +703,20 @@ def _transform_range_iteration(self, node, range_function, reversed=False):
             step_pos = step.pos
             if not isinstance(step.constant_result, _py_int_types):
                 # cannot determine step direction
+                if len(args) == 3:
+                    try:
+                        _fix_range_3_args = []
+                        for arg_node in args:
+                            arg_int = arg_node.coerce_to_integer(self.current_env())
```

Comment:
I see what you mean, but I'm not sure what's the best solution on this. Do you think it's feasible if I limit the arg type to "int" and "long" on this fix for now?